cli/src/tunnels/control_server.rs

```diff
@@ -431,10 +546,14 @@ async fn handle_socket_read(
 			_ = closer.wait() => Err(std::io::Error::new(std::io::ErrorKind::UnexpectedEof, "eof")),
 		}?;
 
+		if read_len == 0 {
```

Comment:
SSH streams errored instead of returning a read length of 0 (idiomatic in rust), so handle this now that we deal with stdin as a possible stream